frontend/src/scenes/experiments/experimentLogic.tsx

```diff
@@ -1021,6 +1044,49 @@ export const experimentLogic = kea<experimentLogicType>([
                 actions.setExperiment(structuredClone(values.unmodifiedExperiment))
             }
         },
+        validateFeatureFlag: async ({ featureFlagKey }: { featureFlagKey: string }, breakpoint) => {
+            await breakpoint(200)
+            const response = await api.get(
+                `api/projects/${values.currentProjectId}/feature_flags/?${toParams({ search: featureFlagKey })}`
+            )
+            const existingErrors = {
+                // :KLUDGE: If there is no name error, we don't want to trigger the 'required' error early
+                name: undefined,
+                ...values.experimentErrors,
+            }
+            if (response.results.length > 0) {
+                const matchingFlag = response.results.find((flag: FeatureFlagType) => flag.key === featureFlagKey)
+                if (matchingFlag) {
+                    let isValid
+                    try {
+                        isValid = featureFlagEligibleForExperiment(matchingFlag)
+                    } catch (error) {
+                        isValid = false
+                    }
+                    actions.setIsExistingFeatureFlag(true)
+                    actions.setIsValidExistingFeatureFlag(isValid)
+                    if (isValid) {
+                        actions.setExperimentManualErrors({
+                            ...existingErrors,
+                            feature_flag_key: null,
+                        })
+                    } else {
+                        actions.setExperimentManualErrors({
+                            ...existingErrors,
+                            feature_flag_key: 'Existing feature flag is not eligible for experiments.',
+                        })
+                    }
+                    return
+                }
+            }
+
+            actions.setIsExistingFeatureFlag(false)
+            actions.setIsValidExistingFeatureFlag(false)
+            actions.setExperimentManualErrors({
+                ...existingErrors,
+                feature_flag_key: validateFeatureFlagKey(featureFlagKey),
+            })
```

Comment:
One problem here: when a new field is focused, the manual errors state is cleared and the error state is reset to _the previous value of experimentErrors_. As such, it's not possible to assign the value to a reducer and then reference the reducer in `errors: ({` because the reducer will contain the value that existed before `validateFeatureFlag()` fired.

Workaround ideas?

**Branch**

![CleanShot 2025-01-31 at 06 22 11](https://github.com/user-attachments/assets/7465250d-747b-4be9-a373-f52f20616b36)

**Master**
![CleanShot 2025-01-31 at 06 24 14](https://github.com/user-attachments/assets/69a71aa9-c7d2-4207-8a81-5d9f2baa9135)

Reply:
> One problem here: when a new field is focused, the manual errors state is cleared and the error state is reset to _the previous value of experimentErrors_.

It actually happens `onBlur` for the field:

https://github.com/keajs/kea-forms/blob/7b7028dae22df4fa624331a713968be6bd8ee2a7/src/components.tsx#L119-L120

https://github.com/keajs/kea-forms/blob/839f6f71b72554ccaf404c5066da5c911b214934/src/builder.ts#L99-L106

Reply:
962bd18d5b (and a302b2af25) seem to work better:

![CleanShot 2025-02-03 at 06 53 51](https://github.com/user-attachments/assets/78848d3d-ccbd-48df-8b20-5d7fa417dbe8)